provider/maas/volumes.go

```diff
@@ -362,8 +362,13 @@ func (mi *maas2Instance) volumes(
 			deviceName := idPath[len(devPrefix):]
 			attachment.DeviceName = deviceName
 		} else if strings.HasPrefix(idPath, devDiskByIdPrefix) {
-			hardwareId := idPath[len(devDiskByIdPrefix):]
```

Comment:
Repeating what I said on IRC: there's no point, since the value that's there would never match any block devices on the host. The worker (diskmanager) that publishes block device info will never put /dev/disk/by-id/www-... in the BlockDevice.HardwareId field.

The ID fields we populate in VolumeInfo and VolumeAttachmentInfo are used to match them up with a block device on the machine that the volume is attached to. If they never match, then the volume attachment will never be presented to the unit that owns the storage.